helm/cluster-aws/templates/cilium-cleanup/job.yaml

```
@@ -0,0 +1,72 @@
+{{- if eq (required "global.connectivity.cilium.ipamMode is required" .Values.global.connectivity.cilium.ipamMode) "eni" }}
+#
+# When working in eni mode and ENIs need to be deleted (e.g. on cluster deletion), Cilium first needs to detach
+# the ENIs from the EC2 instance, and then delete them.
+# If cilium-operator gets killed / removed after it detaches an ENI but before it deletes it, the ENI will be orphaned.
+#
+# This is a possible scenario on cluster deletion. To prevent it, this pre-delete hook will scale down cilium-operator before deleting.
+# This way ENIs remain attached to the EC2 instances and get automatically deleted by AWS upon instance termination.
+#
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "resource.default.name" . }}-cilium-cleanup
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: before-hook-creation
```

Maybe we can delete it if it was successful? 
```suggestion
    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
```